### DISCUSSION TITLE
Report issue with GitHub Codespaces and Dev Containers (VSCode ext)

### DISCUSSION THREAD

**Discussion by snebjorn** [devcontainers/spec#119] [Q&A] [answered]
Is this the right place to report if a devcontainer.json setup doesn't work correctly in [GitHub Codespaces](https://marketplace.visualstudio.com/items?itemName=GitHub.codespaces) and/or [Dev Containers](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers)?

My assumption is that they all rely on https://github.com/devcontainers/cli

**Comment by Chuxel** [comment, marked as answer]
The devcontainers org focuses on tool-agnostic capabilities, which certainly overlaps with implementations like the3 VS Code Dev Containers / Remote - Containers extension and GitHub Codespaces.  The Dev Container CLI is stand-alone, but can clearly affect these two products - however, really the for users of the VS Code Dev Containers and Codespaces has not changed.  That said, here's a breakdown of spots you can provide feedback:

* Issues / feature requests specific to...
    * VS Code Dev Containers / Remote - Containers extension: https://github.com/microsoft/vscode-remote-release
    * GitHub Codespaces: https://github.com/community/community/discussions/categories/codespaces
    * CLI or common implementation of the dev container spec (non-UI): https://github.com/devcontainers/cli
    * Dev Container templates managed by the spec maintainers (often previously in vscode-dev-containers): https://github.com/devcontainer/templates
    * Dev Container Features managed by the spec maintainers (often previously in vscode-dev-containers): https://github.com/devcontainer/features
    * Pre-built images managed by the spec maintainers (often previously in vscode-dev-containers): https://github.com/devcontainer/images
* Feature requests for new dev container metadata related capabilities: this repo

**Reply by snebjorn** [reply]
Thanks for this.

Judging from that I'm guessing that if there's a problem in both GitHub Codespaces and Dev Container then it's an issue with [devcontainers/cli](https://github.com/devcontainers/cli)
